solr/core/src/java/org/apache/solr/search/facet/AvgAgg.java

```diff
@@ -82,6 +84,18 @@ public void merge(Object facetResult, Context mcontext1) {
       sum += numberList.get(1).doubleValue();
     }
 
+    @Override
+    public void readState(JavaBinCodec codec, DataInputInputStream dis, Context mcontext) throws IOException {
+      num = (long) codec.readVal(dis);
+      sum = (double) codec.readVal(dis);
```

Comment:
Meta comment. How does the industry handle versioning JavaBin encoded things? Should we consider beginning the every encoded stream with a version header?

Reply:
Yeah, encoding versions is one option. Depends a little how granular you want to get with all of these entities. For now, I can just add a single top level version that isn't handled in any specific way. Just to have a slot there for the future.

Reply:
An alternative (similar discussion above) is close to what protos does with field numbers where you can cleanly handle missing and/or extra values. But this requires a good bit more involved changes and I don't think is worth it yet.

Reply:
Yeah at that point we might just adopt protos 😄